### DISCUSSION TITLE
Jacoco coverage for a quarkus extension deployment module `@QuarkusUnitTest` not as expected

### DISCUSSION THREAD

**Discussion by HerrDerb** [quarkusio/quarkus#38085] [Q&A]
I have a quarkus extension that has a `runtime` and a `deplyoment` module. The runtime module contains regular jUnit tests and the deployment module contains `@QuarkusUnitTest`s.
When applying the `io.quarkus:quarkus-jacoco` gradle plugin to the project I do get an aggregated coverage of the runtime and the deployment module.

When inspecting the coverage created for the deployment module, I figured that the measured coverage is not what I would have expected. The build processors [deplyoment module]  get a full coverage, also the pojos. But all beans do not get a coverage at all,  although I do inject them into the test and call methods on them.

Is this somehow of a classpath problem as `@QuarkusUnitTest` for the deployment module are kind of different?
Is a special configuration required to get a correct coverage for deployment module tests or is there a technical issue that prevents this at all?
My configuration (root build.gradle) looks like this:
```
          ...
          testImplementation 'io.quarkus:quarkus-jacoco'
        }

        test {
            systemProperty "java.util.logging.manager", "org.jboss.logmanager.LogManager"
            useJUnitPlatform()
                finalizedBy jacocoTestReport

            jacoco {
                destinationFile = layout.buildDirectory.file("./jacoco-quarkus.exec").get().asFile
            }
            jacocoTestReport.enabled = false
        }

        jacocoTestReport {
            reports {
                xml.required = true
                csv.required = false
                html.required = true
            }
        }
```

**Comment by mkouba** [comment]
@HerrDerb did you manage to fix the problem?

**Reply by HerrDerb** [reply]
No sorry, we currently just life with it as it doesn't bother us much 🤷‍♂️

**Reply by mkouba** [reply]
@HerrDerb No problem, I was just curious. Thanks!

@gsmet do you happen to know if there's a way to run Jacoco for extensions?

**Reply by mkouba** [reply]
FYI https://github.com/quarkusio/quarkus/pull/52153
